service/frontend/handler.go

```diff
@@ -391,6 +402,9 @@ func (wh *WorkflowHandler) RecordActivityTaskHeartbeat(
 	sw := wh.startRequestProfile(scope)
 	defer sw.Stop()
 
+	// Count the request in the RPS, but we still accept it even if RPS is exceeded
```

Comment:
Throttling those calls would cause timeouts, which would still put pressure on the system. I understand the concern though, a bug in client could cause it to be making wrong calls excessively and take us down. Unlikely (hopefully), but possible. 
So here's the proposal: Once we add other protections in the system at a more fine-grained level, we can change the frontend throttling to be across all requests (with higher limits) so that it just acts as a final layer of defense in such cases.